docs/src/how_it_works.md

```diff
@@ -0,0 +1,77 @@
+# How it Works (By Example)
+
+## Example Problem
+
+Let $f$ be a function of position ($\bar{r}$)
+```julia
+function f(r̄::Meshes.Point)
+    x, y, z = to(r̄)
+    ...
+end
+```
+
+Let the integration domain be the space (a ball) enclosed by a sphere centered on the origin with a radius of 5 meters.
+```julia
+center = Meshes.Point(0u"m", 0u"m", 0u"m")
+radius = 5.0
+ball = Meshes.Ball(center, radius)
+```
+
+This integral is often expressed abstractly as simply the following, where the triple integral signs and $\text{d}V$ indicate that the integration domain is some three-dimensional volume.
+```math
+\iiint f(\bar{r}) ~ \text{d}V
+```
+
+Integrals like this are often solved manually by selecting an appropriate coordinate system and limits that neatly represent the integration domain, e.g.
+```math
+\int_0^{\pi} \int_0^{2\pi} \int_0^{5} f(\bar{r}) ~ \text{d}\rho~\text{d}\theta~\text{d}\phi
+```
+
+This works great for simple geometries, but requires integration code that is geometry-specific. This package, by contrast, leverages parametric functions defined in Meshes.jl and differential forms to define integral methods that are general solutions for all geometries.
```

Comment:
Sorry for bringing this up again, but is it really "by contrast"? Aren't we essentially doing exactly this? I mean parametric functions are basically "selecting an appropriate coordinate system", aren't they? It's also "geometry-specific" because we need the geometry-specific parametric function (only that this is defined in Meshes.jl). Having a generic interface for the parametric functions (e.g. the fact that they always (well, with exceptions like `Line` etc.) map a $[0, 1]^d$ hypercube) allows us to define integral methods in MeshIntegrals.jl that are general for all geometries. Your limits from the coordinate system $[0, \pi]\times [0, 2\pi]\times [0, 5]$ in the example above is just the same as the $[0, 1]^3$ domain from the parametric function defined in Meshes.jl with a rescaling. The parametric function then tells us how we compute the point $\bar r$ from $(\rho, \theta, \phi)$. For the specific parametrization $g:   [0, 5]\times\ [0, 2\pi]\times [0, \pi] \to B_5((0, 0, 0)), (r, \phi, \theta)^T\mapsto (r \sin(\theta)\cdot\cos(\phi), r\sin(\theta)\cdot\sin(\phi), r\cos(\theta))^T$, ($B_5((0, 0, 0))$ being the ball with center $(0, 0, 0)$ and radius 5) we would also have to take the Jacobian determinant of $r^2\sin(\theta)$ into account, while the rescaled parametrization in Meshes.jl has a slightly different determinant of the Jacobian, but essentially does the same.

Reply:
This is all correct. My intention with the quoted statement was simply to convey that, as an end-user presented with this kind of problem, your basic choices are:
- Derive and write all of that code yourself on a per-geometry basis. Then do it again for every new particular geometry.
- Use MeshIntegrals.jl where we've done the work for you, and it's done using a generalized method that "just works".

For context, a big part of my motivation in writing this package came from a prior effort, [JefimenkoModels.jl](https://github.com/mikeingold/JefimenkoModels.jl), a computational EM package that relies on solving time-domain integral equations of current fields over source domains. This was before I even knew about Meshes.jl, so a lot of that work was essentially duplicative effort trying to implement my own Unitful coordinate systems and geometries and mapping them onto integral solvers. When I have to revisit it, I can pretty much just refactor it to depend on Meshes and MeshIntegrals and the only remaining code is generating the particular integrand equations.

Reply:
Ok, can we maybe just delete the "by contrast" to avoid confusion?

Reply:
No problem. Reading it again, I don't think the "by contrast" adds anything to the statement anyway. I just dropped it in a commit.